unit_tests/UnitTestKokkosME.h

```diff
@@ -133,17 +133,19 @@ class KokkosMEViews
             for (int simdIndex=0; simdIndex < simdElems; ++simdIndex) {
               element = b[bktIndex*simdLen + simdIndex];
               elemNodes[simdIndex] = bulk_.begin_nodes(element);
-              fill_pre_req_data(dataNeeded_, bulk_, AlgTraits::topo_, element,
-                                *prereqData[simdIndex], alsoProcessMEViews);
+              fill_pre_req_data(dataNeeded_, bulk_, element, *prereqData[simdIndex], alsoProcessMEViews);
             }
 
-            copy_and_interleave(prereqData, simdElems, simdPrereqData,
+            copy_and_interleave(prereqData.data(), simdElems, simdPrereqData,
                                 alsoProcessMEViews);
-            fill_master_element_views(dataNeeded_, bulk_, AlgTraits::topo_,
-                                      element, simdPrereqData);
+            fill_master_element_views(dataNeeded_, bulk_, simdPrereqData);
 
             func(simdPrereqData, meSCS_, meSCV_);
           });
+
+          for(int simdIndex=0; simdIndex<simdLen; ++simdIndex) {
```

Comment:
`std::unique_ptr` instead?

Reply:
Shreyas, thanks for the detailed review!
Regarding std::unique_ptr, you're right. I'll use that here.

Regarding putting the 'SharedMemData' struct in a header, I'm not sure whether it will be useful outside of this file. I was just trying to clean up the noise inside the bucket loop. But if we put the loop in a function that takes a lambda (like Robert did in the unit-test-kokkos-me stuff), then perhaps this SharedMemData is something we would pass to the lambda... and in that case it belongs in a header.

Regarding the switch from topo to rank and nodes-per-entity, Stefan and I discussed that on Friday and I may not remember all the details, but it seemed cleaner to only pass in the info that's needed. We ultimately are thinking about different types of algorithms, including face and face/elem algorithms. So we don't want ScratchViews for elem-node gathers to have the face topo, and vice-versa.

Reply:
Right.. We want to minimize "topo" since we may have two floating around for a face/element and face combination.

Is this ready to go? It looks good to me, however, I want to make sure @sayerhs feels good about this push. In general, it is only a prep for things to come. As we learn more, some of these things above may change.

Reply:
@spdomin I am good with this being merged. What is unclear to me is that `ScratchViews` constructor is taking the `nodesPerEntity` instead of `topo`. So is the anticipation that ScratchViews could have multiple `nodesPerEntity` passed to constructor? If not, where is the second `nodesPerEntity` being passed to ScratchViews? My thinking was that Alan had proposed elemViews, faceViews as different ScratchViews instances.

Reply:
That's right, my thinking was that elemViews and faceViews would be separate ScratchViews instances. That could change as we move forward... If they are not separate ScratchViews instances, then we'll have to figure out new arguments to pass in.

Reply:
@alanw0 thanks for confirming. And if `elemViews` and `faceViews` were different instances then they would only have `topo` associated with it, correct? Sorry, for all these questions. I am just getting up to speed with the design that you and Stefan are working on.

Reply:
The more questions the better. I think we want AssembleElemSolver to not have the topo, since it could be managing both face and elem... Since ScratchViews get constructed within AssembleElemSolver, then they can't have topo either. Thus elemViews would get nodesPerElem, and faceViews would get nodesPerFace. 

This is basically thinking out loud, so nothing is in concrete yet.

Reply:
Yes, that is the point. A topo coming into the ElemAlg constructor was only used for ScratchData. However, if you looked at that code, it was only using it for nodesPerElement.